lib/govuk_design_system_formbuilder/traits/localisation.rb

```diff
@@ -4,28 +4,39 @@ module Localisation
     private
 
       def localised_text(context)
-        key = localisation_key(context)
+        return unless @object_name.present? && @attribute_name.present?
 
-        # `I18n.exists?(nil)` returns true when `config.i18n.fallbacks` is
-        # enabled, so only proceed if the key is present too
-        return nil unless key.present? && I18n.exists?(key)
-
-        I18n.translate(key)
+        I18n.translate(localisation_key(context), default: nil) ||
+          I18n.translate(localisation_key(context, html: true), default: nil).try(:html_safe)
       end
 
-      def localisation_key(context)
-        return nil unless @object_name.present? && @attribute_name.present?
-
-        schema(context)
+      def localisation_key(context, html: false)
+        if html
+          schema(context).concat('_html')
+        else
+          schema(context)
+        end
       end
 
       def schema(context)
         schema_root(context)
-          .push(@object_name, @attribute_name)
+          .push(*schema_path)
           .map { |e| e == :__context__ ? context : e }
           .join('.')
       end
 
+      def schema_path
+        [].tap do |path|
+          path.push(@object_name)
+
+          if @value.present?
+            path.push("#{@attribute_name}_options", @value)
+          else
+            path.push(@attribute_name)
+          end
+        end
```

Comment:
I think this can be simplified a little further

```suggestion
        if @value.present?
          [@object_name, "#{@attribute_name}_options", @value]
        else
          [@object_name, @attribute_name]
        end
```

Reply:
Yeah, totally. This was part of an earlier version as I was playing around with the localisation and then I forgot to cleanup the unnecessarily complex code. Thanks for pointing this out!